float32 verticalAngle
float32 horizontalAngle
